src/components/StudentCard.vue

```diff
@@ -1,116 +1,235 @@
+                        :key="assessment.taskId"
+                        class="flex justify-between text-sm align-items-center"
+                      >
+                        <div>{{ taskDisplayNames[assessment.taskId]?.publicName }}</div>
+                        <div>
+                          <PvTag
+                            :severity="getAssessmentStatus(assessment).severity"
+                            :value="getAssessmentStatus(assessment).value"
+                            :icon="getAssessmentStatus(assessment).icon"
+                            class="p-0.5 m-0 font-semibold capitalize"
+                            :style="`min-width: 2rem;`"
+                            rounded
+                          />
+                        </div>
+                      </div>
+                    </div>
+                  </div>
+                </div>
+              </PvAccordionTab>
+            </PvAccordion>
           </div>
         </div>
-      </div>
-    </div>
+      </template>
+    </PvDataView>
   </article>
 </template>
 
 <script setup>
-import { computed } from 'vue';
-import { taskDisplayNames, progressTags } from '@/helpers/reports.js';
+import { computed, ref } from 'vue';
+import { taskDisplayNames, getGradeToDisplay, progressTags } from '@/helpers/reports.js';
 import _capitalize from 'lodash/capitalize';
+import _orderBy from 'lodash/orderBy';
 import PvTag from 'primevue/tag';
 import PvButton from 'primevue/button';
+import PvAccordion from 'primevue/accordion';
+import PvAccordionTab from 'primevue/accordiontab';
+import PvDataView from 'primevue/dataview';
+import PvSelect from 'primevue/select';
+import useUserDataQuery from '@/composables/queries/useUserDataQuery';
+import { useGameStore } from '@/store/game';
+import { storeToRefs } from 'pinia';
+import router from '@/router/index.js';
+
+const gameStore = useGameStore();
+const { selectedAdmin } = storeToRefs(gameStore);
+
+const sortOptions = [
+  { label: 'Name', value: 'name' },
+  { label: 'Date Opened', value: 'dateOpened' },
+  { label: 'Date Closed', value: 'dateClosed' },
+];
+
+const sortKey = ref(sortOptions[0]);
+const sortOrder = ref(1);
+const sortField = ref('name');
+const expanded = ref(false);
+
+const activeIndices = computed(() => {
+  if (!expanded.value) return [];
+  return sortedAssignments.value?.map((_, index) => index) || [];
+});
+
+const onSortChange = (event) => {
+  sortField.value = event.value.value;
+  sortOrder.value = 1;
+};
+
+const setExpand = () => {
+  expanded.value = !expanded.value;
+};
+
+const sortedAssignments = computed(() => {
+  // For dateOpened, we want ascending order (earliest to latest) when sortOrder is 1
+  const order =
+    sortField.value === 'dateOpened'
+      ? sortOrder.value === 1
+        ? 'asc'
+        : 'desc'
+      : sortOrder.value === 1
+      ? 'desc'
+      : 'asc';
+  return _orderBy(Object.values(props.assignments), [sortField.value], [order]);
+});
```

Comment:
```suggestion
const sortedAssignments = computed(() => {
  // For dateOpened, we want ascending order (earliest to latest) when sortOrder is 1
  const isAscending = sortOrder.value === 1;
  const isDateOpened = sortField.value === 'dateOpened';
  const order = (isDateOpened === isAscending) ? 'asc' : 'desc';

  return _orderBy(Object.values(props.assignments), [sortField.value], [order]);
});
```
AI suggestion for refactoring the nested ternary (which can  be difficult to parse).